internal/core/metadata/application/deviceservice.go

```diff
@@ -77,6 +77,11 @@ func PatchDeviceService(dto dtos.UpdateDeviceService, ctx context.Context, dic *
 		"DeviceService patched on DB successfully. Correlation-ID: %s ",
 		correlation.FromContext(ctx),
 	)
+	// Don't invoke callback if only lastConnected field is updated
+	if dto.LastConnected != nil && dto.Description == nil && dto.LastReported == nil &&
+		dto.BaseAddress == nil && dto.AdminState == nil && len(dto.Labels) == 0 {
```

Comment:
It's a good idea to not invoke callback if only lastConnected field is updated.
However, this if statement is hard to maintain if we add more fields into the Device struct.
Please write a helper function and use the reflect package to replace this if statement.
https://pkg.go.dev/reflect
Use a for loop to go through all the fields in this DTO and check the logic you designed.